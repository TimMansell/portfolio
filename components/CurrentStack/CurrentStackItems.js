import React from 'react';
import classnames from 'classnames';

import CurrentStackItem from './CurrentStackItem';

import styles from './CurrentStack.module.scss';

import stackItems from './json/stack.json';

import { useIsFlexGapSupported } from 'hooks/useFlexGap';

export const CurrentStackItems = () => {
  const hasFlexGap = useIsFlexGapSupported();

  const stackClasses = classnames(styles.stack, {
    [styles.flexGap]: hasFlexGap,
  });

  return (
    <div className={stackClasses} data-test="stack-items">
      {stackItems.map((stack, i) => (
        <CurrentStackItem {...stack} key={i} />
      ))}
    </div>
  );
};

export default CurrentStackItems;
