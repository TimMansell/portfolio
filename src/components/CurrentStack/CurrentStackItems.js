import React from 'react';

import CurrentStackItem from './CurrentStackItem';

import styles from './CurrentStack.module.scss';

import stackItems from './json/stack.json';

export const CurrentStackItems = () => {
  return (
    <div className={styles.stack} data-test="stack-items">
      {stackItems.map((stack, i) => (
        <CurrentStackItem {...stack} key={i} />
      ))}
    </div>
  );
};

export default CurrentStackItems;
