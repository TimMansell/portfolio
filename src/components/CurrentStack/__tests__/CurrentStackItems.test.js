import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import CurrentStackItems from '../CurrentStackItems';

jest.mock(
  '../json/stack.json',
  () => [
    {
      name: 'react',
      url: 'url',
      width: '65',
      height: '50',
    },
    {
      name: 'nodejs',
      url: 'url',
      width: '90',
      height: '50',
    },
  ],
  { virtual: true }
);

describe('CurrentStackItems', () => {
  it('should render my component', () => {
    render(<CurrentStackItems />);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<CurrentStackItems />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correct amount of items', () => {
    const { queryAllByTestId } = render(<CurrentStackItems />);
    const list = queryAllByTestId('stack-item');

    expect(list).toHaveLength(2);
  });
});
