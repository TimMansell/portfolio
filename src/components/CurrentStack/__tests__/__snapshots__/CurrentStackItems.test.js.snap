// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentStackItems should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="stack"
    data-test="stack-items"
  >
    <div
      class="item"
      data-testid="stack-item"
    >
      <a
        data-testid="stack-item-link"
        href="url"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="container"
        >
          <img
            alt="react"
            class="image"
            height="50"
            loading="lazy"
            src="[object Object]"
            title="react"
            width="65"
          />
        </div>
      </a>
    </div>
    <div
      class="item"
      data-testid="stack-item"
    >
      <a
        data-testid="stack-item-link"
        href="url"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="container"
        >
          <img
            alt="nodejs"
            class="image"
            height="50"
            loading="lazy"
            src="[object Object]"
            title="nodejs"
            width="90"
          />
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
